http.ts

```diff
@@ -0,0 +1,129 @@
+import {log} from "./util";
+
+import { pubInternal, sub } from "./dispatch";
+import { main as pb } from "./msg.pb";
+
+const servers = new Map<number, HttpServer>();
+const dec = new TextDecoder("utf8");
+const enc = new TextEncoder();
+
+export function initHttp() {
+  sub("http", (payload: Uint8Array) => {
+      const msg = pb.Msg.decode(payload);
+      const id = msg.httpServerId;
+      const s = servers.get(id);
+      s.onMsg(msg);
+  });
+}
+
+export interface RequestOptions {
+    method?: string;
+    referrer?: string;
+    mode?: string;
+    credentials?: string;
+    redirect?: string;
+    integrity?: string;
+    cache?: string;
+}
+
+export class Request {
+    readonly defaultMethod = "GET";
+    path: string;
+    method: string;
+    // tslint:disable-next-line: no-any
+    body: any | string;
+    constructor(url: string, opts?: RequestOptions) {
+      if (opts == null) {
+        opts = {
+            method: this.defaultMethod
+        };
+      }
+      this.method = opts.method;
+    }
+}
+
+export class Response {
+    serverId: number;
+    channel: string;
+    constructor() {
+    }
+    write(data: Uint8Array | string) {
+        let rawData: Uint8Array;
+        if (typeof(data) === "string") {
+            rawData = enc.encode(data);
+        } else {
+            rawData = data as Uint8Array;
+        }
+        pubInternal(this.channel, {
+            command: pb.Msg.Command.HTTP_RES_WRITE,
+            httpResBody: rawData
+        });
+    }
+    status(code: number) {
+        pubInternal(this.channel, {
+            command: pb.Msg.Command.HTTP_RES_STATUS,
+            httpResCode: code
```

Comment:
And do not reuse fields from other commands.

s/httpResCode/httpResHeaderCode/